Src/AutoFixture/DomainName.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+
+namespace Ploeh.AutoFixture
+{
+    /// <summary>
+    /// Represents a domain name.
+    /// </summary>
+    public class DomainName
+    {
+        private string domainName;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DomainName"/> class. Throws ArgumentNullException
+        /// if domainName is null. Throws ArgumentException if domainName is empty.
+        /// </summary>
+        public DomainName(string domainName)
+        {
+            if (domainName == null)
+            {
+                throw new ArgumentNullException("domainName");
+            }
+
+            if (domainName.Length == 0)
```

Comment:
I can't even recall having that discussion related to `EmailAddressLocalPart` :flushed: I wonder why we left it in...